### DISCUSSION TITLE
Power Monitoring System

### DISCUSSION THREAD

**Discussion by pablo-collantes** [Comet-Robotics/URC#26] [PCB/Power]
Can we develop a power monitoring system that can log voltage, current, and power consumption across different components such as pc, radio, and drivetrain. 
Should some of these systems implement short-circuit protections or under/over voltage protections.

**Comment by PablotheEditor** [comment]
So the overall idea right now is to use an Arduino to gather power telemetry from the different components.
- The battery itself can be measured with one of the ADC pins on the Arduino
- it uses a reference voltage on the aref pin
- in order to measure the battery we need a resistive array for a higher impendence input and to reduce the voltage so the board don't fry
attached I've added a diagram of what I'm thinking should work
along with some pretty basic Arduino code that'll need to be updated when we get the actual modules

![IMG_7243](https://github.com/user-attachments/assets/5504b2fe-8e8d-411f-942a-684872930daa)

[DC Voltage Measurement Sketch.txt](https://github.com/user-attachments/files/17357822/DC.Voltage.Measurement.Sketch.txt)
[ACS712 Sensor Sketch.txt](https://github.com/user-attachments/files/17357823/ACS712.Sensor.Sketch.txt)
